src/main/java/blusunrize/immersiveengineering/common/crafting/serializers/BottlingMachineRecipeSerializer.java

```diff
@@ -56,16 +69,21 @@ public BottlingMachineRecipe readFromJson(ResourceLocation recipeId, JsonObject
 	public BottlingMachineRecipe fromNetwork(ResourceLocation recipeId, FriendlyByteBuf buffer)
 	{
 		List<Lazy<ItemStack>> outputs = PacketUtils.readList(buffer, IERecipeSerializer::readLazyStack);
-		Ingredient input = Ingredient.fromNetwork(buffer);
+		int inputCount = buffer.readInt();
+		IngredientWithSize[] ingredients = new IngredientWithSize[inputCount];
+		for(int i = 0; i < ingredients.length; i++)
+			ingredients[i] = IngredientWithSize.read(buffer);
```

Comment:
`PacketUtils::read/writeList`? I think vanilla has a helper for it now as well.

Reply:
read/writeList would give me a list, I need an array. No point in making a list that then needs to be put an array again.